store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/operation/handler/FederatedOperationHandler.java

```diff
@@ -56,13 +59,36 @@ public class FederatedOperationHandler<P extends Operation> implements Operation
 
     @Override
     public Object doOperation(final P operation, final Context context, final Store store) throws OperationException {
+
+        // Check inside operation chains in case there are operations that don't require running on sub graphs
+        if (operation instanceof OperationChain) {
+            List<Operation> ops = ((OperationChain<?>) operation).flatten();
+            List<Class<? extends Operation>> opClasses = ops.stream().map(Operation::getClass).collect(Collectors.toList());
+            // If all the operations in the chain can be handled by the store then execute them
+            if (((FederatedStore) store).getStoreSpecificOperations().containsAll(opClasses)) {
+                return new OperationChainHandler<>(store.getOperationChainValidator(), store.getOperationChainOptimisers())
+                    .doOperation((OperationChain<Object>) operation, context, store);
+            }
+
+            // Check if we have a mix as that is an issue
+            // It's better to keep federated and non federated separate so error and report back
+            for (final Class<? extends Operation> op : opClasses) {
```

Comment:
Could this be simplified to just `containsAny`?

Reply:
`containsAny()` isn't a method on Collections I could probably do this using a `disjoint()` call instead though.

Reply:
Oops yes I think I'm getting confused with streams